File: it/test-versions/expected/js/bar/_2_0_0.js
export class Other {
  constructor(name2) {
    this.name2 = name2;
  }

  static decode(data) {
    const v_name2 = data["name2"];

    if (v_name2 === null || v_name2 === undefined) {
      throw new Error("name2" + ": required field");
    }

    return new Other(v_name2);
  }

  encode() {
    const data = {};

    if (this.name2 === null || this.name2 === undefined) {
      throw new Error("name2: is a required field");
    }

    data["name2"] = this.name2;

    return data;
  }
}

File: it/test-versions/expected/suite/python/bar/_2_0_0.py
class Other:
  def __init__(self, name2):
    self.name2 = name2

  @staticmethod
  def decode(data):
    f_name2 = data["name2"]

    return Other(f_name2)

  def encode(self):
    data = dict()

    if self.name2 is None:
      raise Exception("name2: is a required field")

    data["name2"] = self.name2

    return data

  def __repr__(self):
    return "<Other name2: {!r}>".format(self.name2)
